Library/Homebrew/brew.rb

```diff
@@ -82,7 +82,11 @@
     # `Homebrew::Help.help` never returns, except for unknown commands.
   end
 
-  if internal_cmd || Commands.external_ruby_v2_cmd_path(cmd)
+  if internal_cmd
+    Utils::Analytics.report_command_run(T.must(cmd), args)
```

Comment:
I know this complicates things, but we probably want to validate args first or we could get a lot of junk/typos/spam.

#16815 should however make this much easier as it lifts arg parsing to happen before running the command, rather than it being something the commands themselves do.

Reply:
@Bo98 We're not sending any of the `named_args` here. I was thinking of restricting it further to only e.g. arguments starting with `-` for internal commands and maybe doing something more clever for subcommands.

Reply:
> We're not sending any of the named_args here

Looks like we are because this is before named_args parsing happens:

```console
$ HOMEBREW_ANALYTICS_DEBUG=1 brew info hello
... args="["hello"]" ...
```

```console
$ HOMEBREW_ANALYTICS_DEBUG=1 brew info --junk=key it contains everything on the command line
... args="["--junk=key", "command", "contains", "everything", "it", "line", "on", "the"]" ...
```

We probably don't want flag values either.